### ISSUE TITLE
Diffusion simulations with multiple species involved in reactions

### ISSUE THREAD

**Issue by jgostick** [PMEAL/OpenPNM#1680]
We need to come up with a way to do reactive transport with multiple species in the mixture, so we can have the following source term:  ``r = kC_{A}C_{B}`` where both species A and B are diffusing.

**Comment by ma-sadeghi** [comment]
Indeed. I think we also need to unify/integrate mixtures with phases in parallel. I don't exactly know how mixtures work and they relate to our phases.
